hw/misc/hermes.c

```diff
@@ -966,6 +981,20 @@ static void hermes_instance_init(Object *obj)
 {
     HermesState *hermes = HERMES(obj);
     memcpy(&hermes->bar2, &bar2_init, sizeof(bar2_init));
+
+    for(size_t i = 0; i < H2C_CHANNELS; i++){
+        memcpy(&hermes->bar2.h2c[i], &h2c_init, sizeof(h2c_init));
+        hermes->bar2.h2c[i].irq_vector = 2 * i;
+        hermes->bar2.h2c[i].ch = i;
+        sprintf(hermes->bar2.h2c[i].name, "h2c-dma-%lu", i);
```

Comment:
You need to allocate `name` first. Or change it to a large enough array instead of a pointer.

Reply:
Good catch. I had in my mind that it was a char array.